### PR TITLE
Update contribution guidelines

### PR DESCRIPTION
**Pull request by MarceloSalazar**
As discussed and agreed.

@teetak01 

cc @maclobdell @toyowata

### CODE REVIEW

**Comment by iot-pdmc** [comment]
Thanks for your changes.
@ARMmbed/pdmc-integrators please review.

File: README.md [hunk 1]
```diff
@@ -308,6 +308,46 @@ This should be verified by executing the Pelion E2E python test library tests.
 
 <span class="notes">**Note:** Future version will bring in also firmware update test as part of the minimum test set.</span>.
 
+## Contributing platform support
+
+The contribution of platform support to this repository is restricted to Arm Mbed Partners and Arm Engineering teams. If you’d like to add a custom or community-based platform, please fork this repository and add it into your own account.
+Expectations on contributions:
+
+-	No code changes in `main.cpp`.
+This is a minimal and generic application that’s expected to work on out of the box with all platforms listed in the documentation and [Pelion Quick-start](https://os.mbed.com/guides/connect-device-to-pelion/) guide
```
Comment (JanneKiiskila) [inline README.md]: Dot at the end missing.

File: README.md [hunk 2]
```diff
@@ -308,6 +308,46 @@ This should be verified by executing the Pelion E2E python test library tests.
...
+
+- No changes to the hash of `mbed-os.lib ` or `mbed-cloud-client.lib` files.
+These are maintained and updated by the Arm team.
+
+-	No extra files or `.mbedignore` with removal of Mbed OS code.
```
Comment (JanneKiiskila) [inline README.md]: If such things would be needed, they imply eithe
- lack of configuration (in which case an issue should be raised in [Mbed OS](https://github.com/ARMmbed/mbed-os/issues/) repository) or
- Mbed OS release used by this repo is not update-to-date (in which case you can raise an issue for the `mbed-os.lib` to be updated by the maintainers of this repo).

File: README.md [hunk 3]
```diff
@@ -308,6 +308,46 @@ This should be verified by executing the Pelion E2E python test library tests.
...
+
+-	Configuration (required)
+     - `mbed_app.json` to add components or features. Please follow the guidelines in the porting section of the docs.
+
+-	Drivers (optional)
+     -	If required, drivers for networking or storage (non-default) can be added in the `drivers` folder using an external library (.lib). For example  `COMPONENT_MYDRIVER.lib` and enabling in `mbed_app.json`.
+
+-	Bootloader (required)
+     -	The configuration should be provided in either [mbed-bootloader](https://github.com/ARMmbed/mbed-bootloader) repository (as default configuration) or bootloader folder in this repository (if non-default). Our recommendation is to contribute to the mbed-bootloader repository whether possible. Please indicate where the bootloader configuration lives.
+     -	Binaries should be generated and contributed following the name conventions (see bootloader folder).
```
Comment (JanneKiiskila) [inline README.md]: Link needed.

File: README.md [hunk 4]
```diff
@@ -308,6 +308,46 @@ This should be verified by executing the Pelion E2E python test library tests.
...
+
+-	Indication of platform support
+     -	Please update `README.md` file and add an entry to the list of supported boards.
+
+-	Test results and other information
+     -	Attach test logs for required toolchains as documented [here](https://os.mbed.com/docs/mbed-os/latest/tools/index.html)
+           -	Greentea (Mbed OS tests, including integration tests)
```
Comment (JanneKiiskila) [inline README.md]: Extra spaces (compare to next row).

File: README.md [hunk 5]
```diff
@@ -308,6 +308,46 @@ This should be verified by executing the Pelion E2E python test library tests.
...
+This is a minimal and generic application that’s expected to work on out of the box with all platforms listed in the documentation and [Pelion Quick-start](https://os.mbed.com/guides/connect-device-to-pelion/) guide
+
+- No changes to the hash of `mbed-os.lib ` or `mbed-cloud-client.lib` files.
+These are maintained and updated by the Arm team.
+
+-	No extra files or `.mbedignore` with removal of Mbed OS code.
+
+-	Configuration (required)
+     - `mbed_app.json` to add components or features. Please follow the guidelines in the porting section of the docs.
+
```
Comment (JanneKiiskila) [inline README.md]: Also, 

- You agree that the configuration changes contributed are considered Apache 2.0 licensed open source content.

File: README.md [hunk 6]
```diff
@@ -308,6 +308,46 @@ This should be verified by executing the Pelion E2E python test library tests.
...
+     -	If required, drivers for networking or storage (non-default) can be added in the `drivers` folder using an external library (.lib). For example  `COMPONENT_MYDRIVER.lib` and enabling in `mbed_app.json`.
+
+-	Bootloader (required)
+     -	The configuration should be provided in either [mbed-bootloader](https://github.com/ARMmbed/mbed-bootloader) repository (as default configuration) or bootloader folder in this repository (if non-default). Our recommendation is to contribute to the mbed-bootloader repository whether possible. Please indicate where the bootloader configuration lives.
+     -	Binaries should be generated and contributed following the name conventions (see bootloader folder).
+
+-	Indication of platform support
+     -	Please update `README.md` file and add an entry to the list of supported boards.
+
+-	Test results and other information
```
Comment (yogpan01) [inline README.md]: I think it will be helpful to provide an example template on how to report the results so there can be consistency in reporting.

**Comment by loverdeg-ep** [comment]
@trowbridgec

File: README.md (hunk 6 above)
Comment (MarceloSalazar) [reply README.md]: This would be nice. Perhaps we could create a PR template with this information and tick boxes wherever is required?

**Comment by MarceloSalazar** [comment]
I've amended the doc as per the feedback received - thanks everyone!

Regarding splitting this into CONTRIBUTING.md, I think it would be a good idea, but IMO it would make sense to split other areas as well (e.g. developer guide & porting process).

Anyway, it would be great to proceed with this PR asap and make these other changes separately when appropriate (feel free to send PRs).

File: README.md [hunk 7]
```diff
@@ -328,22 +329,25 @@ These are maintained and updated by the Arm team.
      -	If required, drivers for networking or storage (non-default) can be added in the `drivers` folder using an external library (.lib). For example  `COMPONENT_MYDRIVER.lib` and enabling in `mbed_app.json`.
 
 -	Bootloader (required)
-     -	The configuration should be provided in either [mbed-bootloader](https://github.com/ARMmbed/mbed-bootloader) repository (as default configuration) or bootloader folder in this repository (if non-default). Our recommendation is to contribute to the mbed-bootloader repository whether possible. Please indicate where the bootloader configuration lives.
-     -	Binaries should be generated and contributed following the name conventions (see bootloader folder).
+     -	The configuration should be provided in either [mbed-bootloader](https://github.com/ARMmbed/mbed-bootloader) repository (as default configuration) or [bootloader](https://github.com/ARMmbed/mbed-os-example-pelion/tree/master/bootloader) folder in this repository (if non-default). Our recommendation is to contribute to the mbed-bootloader repository whether possible. Please indicate where the bootloader configuration lives.
```
Comment (JanneKiiskila) [inline README.md]: whether -> whenever

File: README.md [hunk 8]
```diff
@@ -328,22 +329,25 @@ These are maintained and updated by the Arm team.
...
+     -	Binaries should be generated and contributed following the name conventions in the bootloader folder.
 
 -	Indication of platform support
      -	Please update `README.md` file and add an entry to the list of supported boards.
 
 -	Test results and other information
      -	Attach test logs for required toolchains as documented [here](https://os.mbed.com/docs/mbed-os/latest/tools/index.html)
-           -	Greentea (Mbed OS tests, including integration tests) 
+           - Greentea (Mbed OS tests, including integration tests)
```
Comment (JanneKiiskila) [inline README.md]: Dots missing at end of lines (this and next).

File: README.md [hunk 9]
```diff
@@ -308,6 +308,50 @@ This should be verified by executing the Pelion E2E python test library tests.
...
+-	Test results and other information
+     -	Attach test logs for required toolchains as documented [here](https://os.mbed.com/docs/mbed-os/latest/tools/index.html)
+           - Greentea (Mbed OS tests, including integration tests) 
+           - Pelion E2E tests based on pytest
+     -	Mbed OS and Mbed-cloud-client version used during the tests.
+        Note contributions will be accepted only against versions available in the example at that time.
+
+-	Pull-requests are raised against the master branch. The Arm team makes releases regularly.
+
+-	Pelion-Ready. Indicate if a platform is expected to be marked as Pelion-Ready and therefore be added to the Pelion Quick-start
```
Comment (JanneKiiskila) [inline README.md]: platform -> board

**Comment by MarceloSalazar** [comment]
Fixed.
